run-build-functions.sh

```diff
@@ -102,6 +107,33 @@ run_yarn() {
   fi
 
 
+  if [ "$NETLIFY_YARN_WORKSPACES" = "true" ]
+  then
+    echo "NETLIFY_YARN_WORKSPACES feature flag set"
+    # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
+    # to using the global one (which, for now, is alwasy yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
+    if YARN_IGNORE_PATH=1 yarn workspaces info
+    then
+      # Extract all the packages and respective locations. .data will be a JSON object like
+      # {
+      #   "my-package-1": {
+      #     "location": "packages/blog-1",
+      #     "workspaceDependencies": [],
+      #     "mismatchedWorkspaceDependencies": []
+      #   },
+      #   (...)
+      # }
+      # We need to cache all the node_module dirs, or we'll always be installing them on each run
+      local package_locations=($(YARN_IGNORE_PATH=1 yarn --json workspaces info | jq -r '.data | fromjson | to_entries | .[].value.location | @sh'| tr -d \'))
```

Comment:
The idea here is to rely on the `yarn workspaces info --json` to give us the detailed list of locations where packages are located. That way we can cache all the relevant `node_modules` without needing to guess and run random searches.

Reply:
🤔 I wonder if it would be useful to implement a similar check in the new site flow and/or framework-info library. I'm not sure if that's feasible, but it could be helpful for monorepo site setup.

Reply:
I was about to write that it should be relatively simple to achieve it since we could rely directly in the `package.json`, but completely forgot that the `workspaces` entry will [live in the root `package.json`](https://classic.yarnpkg.com/en/docs/workspaces/) which I guess won't be accessible out of the box for the `framework-info`... 😐

We can maybe open an issue in the repo and discuss it over there? CC @verythorough